### PR TITLE
hw3

### PR DESCRIPTION
**Pull request by dima666Sik**
home work 3

### CODE REVIEW

File: hw_3_zd_2.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+const sarah = {
+    name: 'Sarah',
+    job: {
+        name: 'Andersen',
+        hiringDate: '23.07.2020'
+    }
+}
+
+const {
```
Comment (StanislawVictorovich) [inline hw_3_zd_2.js]: don't use global variables. Make destructurization with input params directly inside the function

Comment (Lazyness) [reply hw_3_zd_2.js]: Done

Comment (Lazyness) [reply hw_3_zd_2.js]: Done

Comment (dima666Sik) [reply hw_3_zd_2.js]: Done

File: hw_3_zd_3.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+function deleteDublicate(arr) {
+    arr.forEach(function (item, i) {
+        console.log("indexArr: " + i + ": " + "value: " + item.id);
```
Comment (StanislawVictorovich) [inline hw_3_zd_3.js]: you have 3 cycles inside the function. This is not optimized.

Comment (dima666Sik) [reply hw_3_zd_3.js]: Done

File: hw_3_zd_3.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+function deleteDublicate(arr) {
+    let set = new Set(arr);
+    let result = [];
+
+    for (let i = 0; i < set.size - 1; i++) {
```
Comment (StanislawVictorovich) [inline hw_3_zd_3.js]: You have 2 cycles - this is not optimized algorithm

File: hw_3_zd_2.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+
+function printJobName({
+    name:nameUser,
+    job:{
+        name,
```
Comment (StanislawVictorovich) [inline hw_3_zd_2.js]: this is not correct way of desctructurization. Please return my code and make changes ONLY in input parameters of the function

File: hw_3_zd_3.js (hunk 3 above)
Comment (dima666Sik) [reply hw_3_zd_3.js]: Done

File: hw_3_zd_2.js (hunk 4 above)
Comment (dima666Sik) [reply hw_3_zd_2.js]: Done

**Review by StanislawVictorovich — APPROVED** [review]

**Pull request merged**